### PR TITLE
EOS:18895 - Add optional arguments to config phase to configure openldap and haproxy independently.

### PR DESCRIPTION
**Pull request by amitwac2608**
1) Removed code to make s3haproxtyconfig cli from s3_haproxy_config.py
2) Moved template files to 'scripts/provisioning/templates' directory.

### CODE REVIEW

File: scripts/provisioning/s3_haproxy_config.py [hunk 1]
```diff
@@ -20,22 +20,33 @@
 
 import os
 import sys
-import argparse
 from s3confstore.cortx_s3_confstore import S3CortxConfStore
 
 class S3HaproxyConfig:
-
+  """HAProxy congiration for S3."""
```
Comment (shahidSTX) [inline scripts/provisioning/s3_haproxy_config.py:26]: Please correct typo.

File: scripts/provisioning/_s3_setup [hunk 2]
```diff
@@ -66,7 +68,12 @@ def main():
       
     elif args.command == 'config':
       from configcmd import ConfigCmd
-      ConfigCmd(args.config).process()
+      if args.only_ldap == True:
+        ConfigCmd(args.config).configure_openldap()
+      elif args.only_haproxy == True:
+        ConfigCmd(args.config).configure_haproxy()
+      else:
```
Comment (nileshgovande) [inline scripts/provisioning/_s3_setup:75]: See if this can be done inside config command

**Review by nileshgovande — APPROVED** [review]

**Comment by gowthamchinna** [comment]
run mini-provisioning

**Comment by gowthamchinna** [comment]
start mini-provisioning

**Review by sachitanands — APPROVED** [review]

**Review by vimalraghwani20 — APPROVED** [review]

**Comment by vimalraghwani20** [comment]
I have retrigger pre-merge job : http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Component-Test/job/S3Server-PreMerge-Test/569/console

**Comment by amitwac2608** [comment]
mini-provisioning re-run passed: http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Mini-Provisioning/job/cortx-s3server/643/

**Pull request merged**